service/integration/open_pbs/geopm_power_limit.py

```diff
@@ -0,0 +1,168 @@
+#  Copyright (c) 2015 - 2023, Intel Corporation
+#  SPDX-License-Identifier: BSD-3-Clause
+#
+
+
+import sys
+
+PYTHON_PATHS = [
+        "/usr/lib/python3.6/site-packages",
+        "/usr/lib64/python3.6/site-packages"]
+
+for p in PYTHON_PATHS:
+    if p not in sys.path:
+        sys.path.insert(0, p)
+
+import os
+import json
+
+import pbs
+
+from geopmdpy import pio
+from geopmdpy import system_files
+
+
+_SAVED_CONTROLS_PATH = "/run/geopm-pbs-hooks/SAVE_FILES"
+_SAVED_CONTROLS_FILE = _SAVED_CONTROLS_PATH + "/power-limit-save-control.json"
+_POWER_LIMIT_RESOURCE = "geopm-node-power-limit"
+
+_power_limit_control = {
+        "name": "MSR::PLATFORM_POWER_LIMIT::PLATFORM_POWER_LIMIT",
+        "domain_type": "board",
+        "domain_idx": 0,
+        "setting": None
+    }
+_controls = [
+        {"name": "MSR::PLATFORM_POWER_LIMIT::PL1_TIME_WINDOW",
+         "domain_type": "board",
+         "domain_idx": 0,
+         "setting": 0.013},
+        {"name": "MSR::PLATFORM_POWER_LIMIT::PL1_CLAMP_ENABLE",
+         "domain_type": "board",
+         "domain_idx": 0,
+         "setting": 1},
+        _power_limit_control,
+        {"name": "MSR::PLATFORM_POWER_LIMIT::PL1_LIMIT_ENABLE",
+         "domain_type": "board",
+         "domain_idx": 0,
+         "setting": 1}
+    ]
```

Comment:
I'm wondering if we need to add handling for checking ```MSR::PLATFORM_POWER_LIMIT:LOCK``` to ensure these settings won't silently be ignored...

Reply:
In looking through the code here, Lowren and I proved to ourselves that when write_control() is called, the lock bit for the PLATFORM RAPL MSR will be checked and if set it will throw.  That exception will bubble out to this Python code, and should call into reject_event().